### PR TITLE
add crawl_google_scholar_citation.py 

### PR DESCRIPTION
**Pull request by boyuuuun**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: web_programming/crawl_google_scholar_citation.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        " for ultrahigh areal capacitance,"
+        " volumetric capacitance, and energy density"
+    )
+    journal_name = "Chem. Mater."
+    volume = "30"
+    page = "3979-3990"
+    year = "2018"
+
+    citation = get_citation(create_url(title, journal_name, volume, page, year))
+    print(citation)
```
Comment (cclauss) [inline web_programming/crawl_google_scholar_citation.py:68]:
```suggestion
    params = {
        "title": (
            "Precisely geometry controlled microsupercapacitors for ultrahigh areal "
            "capacitance, volumetric capacitance, and energy density"
        ),
        "journal_name": "Chem. Mater.",
        "volume": 30,
        "page": "3979-3990",
        "year": 2018,
    }
    print(get_citation("http://scholar.google.com/scholar_lookup", params=params))
```

File: web_programming/crawl_google_scholar_citation.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    clean_url = re.sub(tag, "", url)
+    return clean_url
+
+
+def get_citation(url: str) -> str:
+    """
+    Return the citation number.
+    """
+    url = requests.get(url).text
+    soup = BeautifulSoup(url, "html.parser")
```
Comment (cclauss) [inline web_programming/crawl_google_scholar_citation.py:43]:
```suggestion
    soup = BeautifulSoup(requests.get(base_url, params=params).content, "html.parser")
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: web_programming/crawl_google_scholar_citation.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            " for ultrahigh areal capacitance,"
+            " volumetric capacitance, and energy density"
+        ),
+        "journal_name": "Chem. Mater.",
+        "volume": "30",
+        "page": "3979-3990",
+        "year": "2018",
+    }
+
+    print(get_citation("http://scholar.google.com/scholar_lookup?hl=en&", params=params))
```
Comment (cclauss) [inline web_programming/crawl_google_scholar_citation.py:37]:
```suggestion
    print(get_citation("http://scholar.google.com/scholar_lookup&", params=params))
```

Comment (boyuuuun) [reply web_programming/crawl_google_scholar_citation.py:37]: If '?hl=en&' is dropped, the word 'citation' is automatically translated to the language of the world.. Could you recommend about it?

File: web_programming/crawl_google_scholar_citation.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    params = {
+        "title": (
+            "Precisely geometry controlled microsupercapacitors"
+            " for ultrahigh areal capacitance,"
+            " volumetric capacitance, and energy density"
+        ),
+        "journal_name": "Chem. Mater.",
+        "volume": "30",
+        "page": "3979-3990",
+        "year": "2018",
```
Comment (cclauss) [inline web_programming/crawl_google_scholar_citation.py:34]:
```suggestion
        "year": "2018",
        "hl": "en",
```

Comment (boyuuuun) [reply web_programming/crawl_google_scholar_citation.py:32]: Thanks for the advice!

**Review by cclauss — APPROVED** [review]
Looks really slick!  Thanks for your persistence.

**Pull request merged**